### PR TITLE
Datalayer/build

### PR DESCRIPTION
**Pull request by ShuFFel**

### CODE REVIEW

File: data/src/main/java/entities/JPAPOIEntity.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    @Id
+    @Column(name = "id", length = 15, nullable = false)
+    private long id;
+
+    @Column(name = "name")
+    private String name;
+
+    @Column(name = "X-crd")
+    private double X;
```
Comment (mx510ss) [inline data/src/main/java/entities/JPAPOIEntity.java]: variables & parameters in java should be in lower Case - please change in all places

File: data/src/main/java/entities/JPAPOIEntity.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    @Column(name = "Y-crd")
+    private double Y;
+
+    @Column(name = "Type")
+    @Enumerated(EnumType.STRING)
+    private POITypes type;
+
+    @Column(name = "Discription")
```
Comment (mx510ss) [inline data/src/main/java/entities/JPAPOIEntity.java]: Columns in database should start from lower case also - please fix in all places

File: data/src/main/java/repository/JPAPOIAdapter.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        List<POI> resultList = new ArrayList<POI>();
+        for(JPAPOIEntity i : list){
+            POI tmp = i.toModel();
+            resultList.add(tmp);
+        }
+        return resultList;
+    }
+
+    public void save(Collection<POI> poi) {
+        List<JPAPOIEntity> list = new ArrayList<JPAPOIEntity>();
```
Comment (mx510ss) [inline data/src/main/java/repository/JPAPOIAdapter.java]: new ArrayList<JPAPOIEntity>(poi.size());

**Review by mx510ss — CHANGES_REQUESTED** [review]
Few small changes required

File: data/src/resource/application.properties [hunk 4]
```diff
@@ -0,0 +1,6 @@
+dataSource.driverClassName=org.postgresql.Driver
```
Comment (mx510ss) [inline data/src/resource/application.properties]: this file can be deleted

File: data/src/main/java/repository/JPAPOIAdapter.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Autowired
+    public JPAPOIAdapter(JPAPOIRepository repository){
+        this.repository = repository;
+    }
+
+
+    public List<POI> getByBounds(double X1, double Y1, double X2, double Y2, POITypes type) {
+        List<JPAPOIEntity> list = repository.getByBounds(X1, Y1, X2, Y2, type);
+        List<POI> resultList = new ArrayList<POI>();
```
Comment (mx510ss) [inline data/src/main/java/repository/JPAPOIAdapter.java]: new ArrayList<POI>(list.size())

File: data/src/main/java/repository/JPAPOIAdapter.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        for(JPAPOIEntity i : list){
+            POI tmp = i.toModel();
+            resultList.add(tmp);
```
Comment (mx510ss) [inline data/src/main/java/repository/JPAPOIAdapter.java]: it's can be done in one line:
`resultList.add(i.toModel())`

File: data/src/main/java/repository/JPAPOIAdapter.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class JPAPOIAdapter implements POIRepository {
+
+    private JPAPOIRepository repository;
+
+    @Autowired
```
Comment (mx510ss) [inline data/src/main/java/repository/JPAPOIAdapter.java]: don't need Autowired annotation in this module

File: data/src/main/java/repository/JPAPOIAdapter.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+            resultList.add(tmp);
+        }
+        return resultList;
+    }
+
+    public void save(Collection<POI> poi) {
+        List<JPAPOIEntity> list = new ArrayList<JPAPOIEntity>(poi.size());
+        for(POI i : poi){
+            JPAPOIEntity tmp = new JPAPOIEntity(i);
+            list.add(tmp);
```
Comment (mx510ss) [inline data/src/main/java/repository/JPAPOIAdapter.java]: Also please do it in one line - save a little memory)))

File: data/src/main/java/entities/JPAPOIEntity.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
+package entities;
+
+import enums.POITypes;
+
+import javax.persistence.*;
+
+
+@Entity
+@Table(name = "Object")
```
Comment (mx510ss) [inline data/src/main/java/entities/JPAPOIEntity.java]: Why table called Object? in code we have POI - maybe it's better to have the same names in DB and code

**Review by mx510ss — CHANGES_REQUESTED** [review]
Few more changes required

**Review by mx510ss — APPROVED** [review]
reviewed

File: data/src/main/java/entities/JPAPOIEntity.java (hunk 1 above)
Comment (lennee) [reply data/src/main/java/entities/JPAPOIEntity.java]: Addressed in 7f918919e: RestaurantConfigUpdateProcessor.kt:158 - changed consistentReads=false to true for gating read

Comment (lennee) [reply data/src/main/java/entities/JPAPOIEntity.java]: Fixed in 7f918919e: Changed `consistentReads=false` to `consistentReads=true` in RestaurantConfigUpdateProcessor.kt line 158 for the gating read before write.

**Pull request merged**